astropy/modeling/mappings.py

```diff
@@ -47,31 +47,34 @@ class Mapping(FittableModel):
     linear = True  # FittableModel is non-linear by default
 
     def __init__(self, mapping, n_inputs=None, name=None, meta=None):
+
+        if n_inputs is None:
+            self._n_inputs = max(mapping) + 1
+        else:
+            self._n_inputs = n_inputs
+
+        self._n_outputs = len(mapping)
+        super().__init__(name=name, meta=meta)
+
         if n_inputs is None:
-            self._inputs = tuple('x' + str(idx)
-                                 for idx in range(max(mapping) + 1))
+            self.inputs = tuple('x' + str(idx)
+                                for idx in range(max(mapping) + 1))
+            self.outputs = tuple('x' + str(idx) for idx in range(len(mapping)))
         else:
-            self._inputs = tuple('x' + str(idx)
+            self.inputs = tuple('x' + str(idx)
                                  for idx in range(n_inputs))
```

Comment:
Could this if/else block code be merged with the previous since the logical test is the same?

Reply:
The two mapping models are special in that `n_inputs` and `n_outputs` are determined dynamically when the model is initialized. However, they need to be defined before `super` is called. `inputs` and `outputs` need to be defined after `super` is called so that they can overwrite any defaults.